app/build.gradle

```diff
@@ -12,8 +12,43 @@ android {
     }
     buildTypes {
         release {
-            minifyEnabled false
+            minifyEnabled true
+            shrinkResources true
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            buildConfigField "String", "SERVER_PREFIX", "\"\""
+        }
+
+        debug {
+            buildConfigField "String", "SERVER_PREFIX", "\"test.\""
+        }
+    }
+    flavorDimensions "version"
+
+    productFlavors {
+        free {
+            dimension "version"
+            applicationIdSuffix ".free"
+            versionNameSuffix "-free"
+        }
+
+        paid {
+            dimension "version"
+            applicationIdSuffix ".paid"
+            versionNameSuffix "-paid"
+        }
+    }
+
+    sourceSets {
+        free {
+            java {
+                srcDirs = ['src/free/java']
```

Comment:
actually it's not required. Flavors work from box this way. You can remove scrDirs for free and paid